extensions/applicationinsights-dependencies-js/src/ajax.ts

```diff
@@ -548,6 +551,32 @@ export class AjaxMonitor extends BaseTelemetryPlugin implements IDependenciesPlu
             function _isDisabledRequest(xhr?: XMLHttpRequestInstrumented, request?: Request | string, init?: RequestInit) {
                 let isDisabled = false;
                 let theUrl:string = ((!isString(request) ? ((request ||{}) as Request).url || "" : request as string) ||"").toLowerCase();
+
+                // check excludeRequestFromAutoTrackingPatterns before stripping off any query string
+                arrForEach(_excludeRequestFromAutoTrackingPatterns, (regex: string | RegExp) => {
+                    let isRegExp = false;
+                    if (!isString(regex)) {
```

Comment:
If it's a regex we should just reuse it rather than pull it apart and recreate it
```
let theRegEx = regex;
if (isString(regex)) {
    theRegEx = new RegEx(regex);
}

isDisabled = theRegEx.test(theUrl);

Reply:
Ohhhhh...!!! I just made things more complicated... 😅